plugins/modules/site_workflow_manager.py

```diff
@@ -1625,6 +1624,45 @@ def change_payload_data(self, config):
             self.msg = "Unable to process the payload data : {}".format(str(e))
             self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
 
+    def is_site_exist(self, site_name):
+        """
+        Checks if a site exists in Cisco Catalyst Center by retrieving site information based on the provided site name.
+
+        Args:
+            site_name (str): The name or hierarchy of the site to be retrieved.
+
+        Returns:
+            tuple (bool, str or None): A tuple containing:
+                1. A boolean indicating whether the site exists (True if found, False otherwise).
+                2. The site ID (str) if the site exists, or None if the site does not exist or an error occurs.
+
+        Details:
+            - Calls `get_site()` to retrieve site details from Cisco Catalyst Center.
+            - If the site exists, its ID is extracted from the response, and the function returns (True, site ID).
+            - If the site does not exist, it returns (False, None) and logs an informational message.
+            - Logs detailed debug information about the retrieval attempt and any errors that occur.
+
+        """
+        site_exists = None
+        try:
+            response = self.get_site(site_name)
+
+            if response is None:
+                self.log("No site details retrieved for site name: {0}".format(site_name), "DEBUG")
+                return site_exists
```

Comment:
You are returning NULL here. But in line number 1655, you are returning True.  Please ensure consistency in what you are returning.